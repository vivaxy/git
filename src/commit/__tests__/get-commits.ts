/**
 * @since 2020-02-05 09:43
 * @author vivaxy
 */
import * as path from 'path';
import getCommits from '../get-commits';

const cwd = path.join(__dirname, '..', '..', '..');

test('should get all commits', async function() {
  const commits = await getCommits({
    from: '95331f83d232342eadf764f53c4938b345a99b36',
    to: '5ea8c7e951f8d31f928e78ece3c028762adcf7b6',
    cwd,
  });
  expect(commits).toMatchSnapshot();
});

test('should get commits with paths', async function() {
  const commits = await getCommits({
    cwd,
    paths: ['README.md', 'LICENSE'],
  });
  expect(commits).toMatchSnapshot();
});

test('should get multiple tags', async function() {
  const commits = await getCommits({
    cwd,
    from: '99f0407c99a7bbcd5ad1c194810fe14092e57da3',
    to: 'ebc8de109e22a4546b378bdd90513c686e3a695a',
  });
  expect(commits).toMatchSnapshot();
});
