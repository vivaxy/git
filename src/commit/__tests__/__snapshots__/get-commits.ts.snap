// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get all commits 1`] = `
Array [
  Object {
    "date": 1579502284000,
    "files": Array [
      Object {
        "diffType": "M",
        "filename": "CHANGELOG.md",
      },
      Object {
        "diffType": "M",
        "filename": "package.json",
      },
    ],
    "hash": "5ea8c7e951f8d31f928e78ece3c028762adcf7b6",
    "message": "chore(release): 1.2.0
",
    "tags": Array [
      "v1.2.0",
    ],
  },
  Object {
    "date": 1579502255000,
    "files": Array [
      Object {
        "diffType": "M",
        "filename": "src/index.ts",
      },
      Object {
        "diffType": "A",
        "filename": "src/init/index.ts",
      },
    ],
    "hash": "8e3f52a10d45b3387f02b442c74370fcab55252d",
    "message": "feat(init): :sparkles: add \`git init\`
",
    "tags": Array [],
  },
  Object {
    "date": 1577583012000,
    "files": Array [],
    "hash": "68727e0ec0817e6697465ac97481eb8c46c5b7f3",
    "message": "chore(deps): bump handlebars from 4.1.2 to 4.5.3 (#4)

chore(deps): bump handlebars from 4.1.2 to 4.5.3",
    "tags": Array [],
  },
  Object {
    "date": 1577568329000,
    "files": Array [
      Object {
        "diffType": "M",
        "filename": "yarn.lock",
      },
    ],
    "hash": "2577f00d72a5e9c5b739923a18195438e4ee6d73",
    "message": "chore(deps): bump handlebars from 4.1.2 to 4.5.3

Bumps [handlebars](https://github.com/wycats/handlebars.js) from 4.1.2 to 4.5.3.
- [Release notes](https://github.com/wycats/handlebars.js/releases)
- [Changelog](https://github.com/wycats/handlebars.js/blob/master/release-notes.md)
- [Commits](https://github.com/wycats/handlebars.js/compare/v4.1.2...v4.5.3)

Signed-off-by: dependabot[bot] <support@github.com>",
    "tags": Array [],
  },
]
`;

exports[`should get commits with paths 1`] = `
Array [
  Object {
    "date": 1564533329000,
    "files": Array [
      Object {
        "diffType": "M",
        "filename": "README.md",
      },
    ],
    "hash": "c98d8af0a5b7bc60c481560d1062540d5541719d",
    "message": "refactor(rename): :recycle: rename package from git-info to git
",
    "tags": Array [],
  },
  Object {
    "date": 1564533134000,
    "files": Array [
      Object {
        "diffType": "M",
        "filename": "LICENSE",
      },
      Object {
        "diffType": "M",
        "filename": "README.md",
      },
    ],
    "hash": "e48e65f2b6b6cca868fe26a37c36f4fd39801d1f",
    "message": "refactor(reinit): :recycle: init
",
    "tags": Array [],
  },
  Object {
    "date": 1493120713000,
    "files": Array [
      Object {
        "diffType": "M",
        "filename": "README.md",
      },
    ],
    "hash": "ca9b2da1cf72102d24fa931d975f76457fa8ce7d",
    "message": "feat(DEPRECATED): :sparkles:
",
    "tags": Array [],
  },
  Object {
    "date": 1491913173000,
    "files": Array [
      Object {
        "diffType": "M",
        "filename": "README.md",
      },
    ],
    "hash": "0c56c9c84ed492409871f2efceabd919856af10f",
    "message": "docs(README.md): :memo:
",
    "tags": Array [],
  },
  Object {
    "date": 1491912947000,
    "files": Array [
      Object {
        "diffType": "M",
        "filename": "README.md",
      },
    ],
    "hash": "98657405ee43106ecfcdd2a2cf1b4a79ae6e62dd",
    "message": "feat: :tada:First Commit
",
    "tags": Array [],
  },
  Object {
    "date": 1491912645000,
    "files": Array [
      Object {
        "diffType": "A",
        "filename": "LICENSE",
      },
      Object {
        "diffType": "A",
        "filename": "README.md",
      },
    ],
    "hash": "85c03262afbaf774d45bc9a951bc22445458ccb4",
    "message": "Initial commit",
    "tags": Array [],
  },
]
`;

exports[`should get multiple tags 1`] = `
Array [
  Object {
    "date": 1491913097000,
    "files": Array [
      Object {
        "diffType": "A",
        "filename": "CHANGELOG.md",
      },
      Object {
        "diffType": "M",
        "filename": "package.json",
      },
    ],
    "hash": "ebc8de109e22a4546b378bdd90513c686e3a695a",
    "message": "chore(release): 0.1.0
",
    "tags": Array [
      "v0.1.0",
      "lightweight-tag-1",
      "annotated-tag-1",
    ],
  },
]
`;
